bpf/generictracer/protocol_kafka.h

```diff
@@ -0,0 +1,443 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <bpfcore/vmlinux.h>
+#include <bpfcore/bpf_endian.h>
+#include <bpfcore/bpf_helpers.h>
+#include <bpfcore/utils.h>
+
+#include <common/common.h>
+#include <common/connection_info.h>
+#include <common/large_buffers.h>
+#include <common/http_types.h>
+#include <common/pin_internal.h>
+#include <common/ringbuf.h>
+#include <common/runtime.h>
+#include <common/tp_info.h>
+#include <common/trace_common.h>
+
+#include <generictracer/protocol_common.h>
+#include <generictracer/k_tracer_tailcall.h>
+
+#include <generictracer/maps/protocol_cache.h>
+
+#include <maps/active_ssl_connections.h>
+
+// message_size -> https://kafka.apache.org/protocol#protocol_common
+// The message_size field in the Kafka protocol defines the size of the
+// request/response payload excluding the 4 bytes used by the message_size field itself.
+
+// Every kafka api packet is prefixed by an header
+// https://kafka.apache.org/protocol#protocol_messages
+struct kafka_request_hdr {
+    s32 message_size;
+    s16 request_api_key;     // The API key of this request
+    s16 request_api_version; // The API version of this request
+    s32 correlation_id;      // The correlation ID of this request
+    // client-id is a nullable string
+};
+
+struct kafka_response_hdr {
+    s32 message_size;
+    s32 correlation_id; // The correlation ID of this response
+};
+
+struct kafka_state_data {
+    s32 message_size;
+};
+
+struct kafka_state_key {
+    connection_info_t conn;
+    u8 direction;
+    u8 _pad[3];
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __type(key, connection_info_t);
```

Comment:
Should this be `kafka_state_key`?

```suggestion
    __type(key, kafka_state_key);
```